js/packages/openinference-instrumentation-llama-index/test/llamaIndex.test.ts

```diff
@@ -202,3 +211,122 @@ describe("LlamaIndexInstrumentation", () => {
     });
   });
 });
+
+/*
+ * Tests for embeddings only
+ */
+describe("LlamaIndexInstrumentation - Embeddings", () => {
+  const memoryExporter = new InMemorySpanExporter();
+  const spanProcessor = new SimpleSpanProcessor(memoryExporter);
+  instrumentation.setTracerProvider(tracerProvider);
+
+  tracerProvider.addSpanProcessor(spanProcessor);
+  // @ts-expect-error the moduleExports property is private. This is needed to make the test work with auto-mocking
+  instrumentation._modules[0].moduleExports = llamaindex;
+
+  let openAIQueryEmbedSpy: jest.SpyInstance;
+  let openAITextEmbedSpy: jest.SpyInstance;
+  let openAISpy: jest.SpyInstance;
+
+  jest.mock("llamaindex", () => {
+    const originalModule = jest.requireActual("llamaindex");
+    return {
+      ...originalModule,
+      OpenAIEmbedding: jest.fn().mockImplementation(() => ({
+        getQueryEmbedding:
+          originalModule.OpenAIEmbedding.prototype.getQueryEmbedding,
+      })),
+    };
+  });
+
+  beforeAll(() => {
+    instrumentation.enable();
+  });
+  afterAll(() => {
+    instrumentation.disable();
+  });
+  beforeEach(() => {
+    memoryExporter.reset();
+
+    // Use OpenAI and mock out the calls
+    const response: llamaindex.ChatResponse<llamaindex.ToolCallLLMMessageOptions> =
+      {
+        message: {
+          content: DUMMY_RESPONSE,
+          role: "assistant",
+        },
+        raw: null,
+      };
+
+    // Mock out the chat completions endpoint
+    openAISpy = jest
+      .spyOn(llamaindex.OpenAI.prototype, "chat")
+      .mockImplementation(() => {
+        return Promise.resolve(response);
+      });
+
+    // Mock out the embeddings endpoint
+    openAITextEmbedSpy = jest
+      .spyOn(llamaindex.OpenAIEmbedding.prototype, "getTextEmbeddings")
+      .mockImplementation(() => {
+        return Promise.resolve([
+          FAKE_EMBEDDING,
+          FAKE_EMBEDDING,
+          FAKE_EMBEDDING,
+        ]);
+      });
+
+    openAIQueryEmbedSpy = jest
+      .spyOn(llamaindex.OpenAIEmbedding.prototype, "getQueryEmbedding")
+      .mockImplementation(() => {
+        return Promise.resolve(FAKE_EMBEDDING);
+      });
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+    openAISpy.mockRestore();
+    openAITextEmbedSpy.mockRestore();
+    openAIQueryEmbedSpy.mockRestore();
```

Comment:
you can call jest.restoreAllMocks / jest.resetModules or something i can't remember the exact names but those two instead of restoring each individually